python/tank/platform/bundle.py

```diff
@@ -663,11 +663,49 @@ def create_hook_instance(self, hook_expression):
 
         .. note:: For more information about hook syntax, see :class:`~sgtk.Hook`
 
+        An optional `base_class` can be provided to override the default ``Hook``
+        base class. This is useful for bundles that create hook instances at
+        execution time and wish to provide default implementation without the need
+        to configure the base hook.
+
+        :param hook_expression: Path to hook to execute. See above for syntax details.
+        :param base_class: A python class to use as the base class for the created
+            hook. This will override the default hook base class, ``Hook``.
+        :returns: :class:`Hook` instance.
+        """
+        resolved_hook_paths = self.__resolve_hook_expression(None, hook_expression)
+        return hook.create_hook_instance(
+            resolved_hook_paths,
+            self,
+            base_class=base_class
+        )
+
+    def get_hook_class(self, hook_expression, base_class=None):
+        """
+        Returns the class for the hook resolved for the given expression.
```

Comment:
what is the workflow where this is typically needed? An example in the docstring would help greatly i think.

Reply:
added.